File: DSA-JAVASCRIPT/Data-structures/Array-structure/reverse_array.js
function main() {
    let first, last, reversse;
    arr = [1, 2, 3, 4, 5, 6];
    first = 0;
    last = arr.length - 1;
    reverse = reverseArray(arr, first, last);
    console.log(reverse)
}

function reverseArray(arr, start, end) {
    while (start < end) {
        temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp
        start++;
        end--;
    }
    return arr;

}

main();

File: DSA-PYTHON/Data-structures/Array-structure/reverse_array.py
def main():
    arr = [1, 2, 3, 4, 5, 6]
    first = 0
    last = len(arr) - 1
    reverse = reverse_array(arr, first, last)
    print(reverse)


def reverse_array(arr, start, end):

    while start < end:
        arr[start], arr[end] = arr[end], arr[start]
        start += 1
        end -= 1
    return arr


main()
